src/features/HelpAndKnowledge/SolutionsHelp/SolutionDetails/Solutions/Generic/_components/ContractorForm/index.tsx

```diff
@@ -125,8 +125,9 @@ const ContractorForm = ({
           closeModal();
         }
       })
-      .catch(() => {
-        setHasMutationError(true);
+      .catch(error => {
+        const duplicateError = error.message.includes('duplicate');
```

Comment:
My only suggestion (definitely dont need to take it as this works) but a more accurate and specific identifier to key on would be `uniq_user_id_per_solution_key`, which is present in the gql error message

Reply:
I like that better! Ty!